### ISSUE TITLE
Flooding map in various locations in Norway

### ISSUE THREAD

**Issue by dafadllyn** [osmandapp/OsmAnd#21627]
### Description

The rendering of the coast (sea vs. land) is broken at various places in Norway (see examples below): part of the land is rendered as sea and vice versa. The problem is zoom-dependent, i.e. it doesn't appear (the same) at all zoom levels. It looks like #14857.

### Examples

[Tromsø](https://www.openstreetmap.org/#map=14/69.64866/18.98377):

<p><img src="https://github.com/user-attachments/assets/d3d6efe2-5fb3-4d68-ab0e-a7b069951fca" width="300" /></p>

[Molde](https://www.openstreetmap.org/#map=15/62.73766/7.15892):

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/da7967db-01d9-4e80-b635-956075a79f1e"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/bf21a10f-ba35-487a-ac69-0ca784524f12"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/118ac920-37dc-4891-9091-d6537bc0b8c9"  width="300" height="auto"/> </th>
  </tr>
</table>  

[Nesna](https://www.openstreetmap.org/#map=14/66.19980/13.01880):

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/1fdc9c44-a4e1-49d1-9a1e-417ae3aeab0a"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/299bc6b3-93ed-4aa3-85d7-79e4bda736af"  width="300" height="auto"/> </th>
  </tr>
</table>   

### Steps to reproduce

See description

### Actual result

See description

### Expected result

See description

### Your Environment (required)

```
OsmAnd Version: 4.9.10
Android/iOS version: 15
Device model: Google Pixel 6a
Crash-Logs: –
```

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced . To confirm, here are the steps we followed:  
1. Download the maps for **Tromsø**, **Molde**, and **Nesna** in the OsmAnd app.  
2. Zoom in and out at various levels in these areas.  

**Observed Results**:  
- Certain sections of the land appear flooded.  
- The coastline is displayed with unusual geometric angles at specific zoom levels (3 scr).  


OsmAnd~ 5.0.0#4003m, released: 2024-12-26

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/d8c4cc9b-6652-464c-82db-a42f12b2e71f"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/e73a1f94-cd8d-4675-8070-c2dffff7025d"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/user-attachments/assets/a163b0e8-2076-4f1c-bfa1-05cd0825a8eb"  width="300" height="auto"/> </th>
  <th><img src="https://github.com/user-attachments/assets/3aebc0cd-ccc8-4096-9f06-a904289e2e3c"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by xmd5a2** [comment]
Reproduced. There is no errors in OSM like wrong coastline and our simplified basemap coastline doesn't contain errors in these areas.

**Comment by Andrewschenko** [comment]
The reason of this issue is the same as in [#19240](https://github.com/osmandapp/OsmAnd/issues/19240)

**Comment by Zirochkabila** [comment]
`OsmAnd~ 5.0.0#4136m, Released: 2025-01-15`
`4.9.3.4941`

@Andrewschenko Is this behavior related to the one that was fixed?

`44.33539° N, 8.82499° E`
iOS - same

https://github.com/user-attachments/assets/18b9cb0b-b896-4591-a96e-4bfcf70bfdbf

**Comment by Andrewschenko** [comment]
> `OsmAnd~ 5.0.0#4136m, Released: 2025-01-15` `4.9.3.4941`
> 
> @Andrewschenko Is this behavior related to the one that was fixed?
> 
> `44.33539° N, 8.82499° E` iOS - same
> 
>  2025-01-15.15.49.33.mp4

No. It's the same as the last case in [issue #3821](https://github.com/osmandapp/OsmAnd-iOS/issues/3821)
